src/SearchParameters/index.js

```diff
@@ -103,13 +105,19 @@ var SearchParameters = function( newParameters ) {
 
 SearchParameters.prototype = {
   constructor : SearchParameters,
-  clearRefinements : function clearRefinements() {
-    return this.mutateMe( function( m ) {
-      m.facetsRefinements = {};
-      m.facetsExcludes = {};
-      m.disjunctiveFacetsRefinements = {};
-      m.numericRefinements = {};
-    } );
+
+  /**
+   * Remove all refinements (disjunctive + conjunctive + excludes + numeric filters)
+   * @method
+   * @param {string} [name] - If given, name of the facet / attribute on which  we want to remove all refinements
+   * @return {AlgoliaSearchHelper}
+   */
+  clearRefinements : function clearRefinements( name ) {
+    return this
+      .clearNumericRefinements( name )
+      .clearFacetRefinements( name )
+      .clearExcludeRefinements( name )
+      .clearDisjunctiveFacetRefinements( name );
```

Comment:
Even though it's a good idea to reuse code, I wouldn't do that. mutateMe has a limitation, it always create a new instance, and because it is used in all those 4 methods, it will create sequentially 4 instances. It would be nice to revert that imho.